packages/jupyter-ai-magics/jupyter_ai_magics/partner_providers/aws.py

```diff
@@ -0,0 +1,167 @@
+import copy
+import json
+from typing import Any, Coroutine, Dict
+
+from jsonpath_ng import parse
+from langchain_aws import Bedrock, BedrockEmbeddings, ChatBedrock, SagemakerEndpoint
+from langchain_aws.llms.sagemaker_endpoint import LLMContentHandler
+from langchain_core.outputs import LLMResult
+
+from ..embedding_providers import BaseEmbeddingsProvider
+from ..providers import AwsAuthStrategy, BaseProvider, MultilineTextField, TextField
+
+
+# See model ID list here: https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html
+class BedrockProvider(BaseProvider, Bedrock):
+    id = "bedrock"
+    name = "Amazon Bedrock"
+    models = [
+        "amazon.titan-text-express-v1",
+        "amazon.titan-text-lite-v1",
+        "ai21.j2-ultra-v1",
+        "ai21.j2-mid-v1",
+        "cohere.command-light-text-v14",
+        "cohere.command-text-v14",
+        "cohere.command-r-v1:0",
+        "cohere.command-r-plus-v1:0",
+        "meta.llama2-13b-chat-v1",
+        "meta.llama2-70b-chat-v1",
+        "meta.llama3-8b-instruct-v1:0",
+        "meta.llama3-70b-instruct-v1:0",
+        "meta.llama3-1-8b-instruct-v1:0",
+        "meta.llama3-1-70b-instruct-v1:0",
+        "mistral.mistral-7b-instruct-v0:2",
+        "mistral.mixtral-8x7b-instruct-v0:1",
+        "mistral.mistral-large-2402-v1:0",
+    ]
+    model_id_key = "model_id"
+    pypi_package_deps = ["langchain-aws"]
+    auth_strategy = AwsAuthStrategy()
+    fields = [
+        TextField(
+            key="credentials_profile_name",
+            label="AWS profile (optional)",
+            format="text",
+        ),
+        TextField(key="region_name", label="Region name (optional)", format="text"),
+    ]
+
+    async def _acall(self, *args, **kwargs) -> Coroutine[Any, Any, str]:
+        return await self._call_in_executor(*args, **kwargs)
+
+
+# See model ID list here: https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html
+class BedrockChatProvider(BaseProvider, ChatBedrock):
+    id = "bedrock-chat"
+    name = "Amazon Bedrock Chat"
+    models = [
+        "anthropic.claude-v2",
+        "anthropic.claude-v2:1",
+        "anthropic.claude-instant-v1",
+        "anthropic.claude-3-sonnet-20240229-v1:0",
+        "anthropic.claude-3-haiku-20240307-v1:0",
+        "anthropic.claude-3-opus-20240229-v1:0",
+        "anthropic.claude-3-5-sonnet-20240620-v1:0",
```

Comment:
Titan, Meta, Mistral should also be listed here. For Cohere support, you need to pass in an additional `beta_use_converse_api`.  Can add these in a separate PR though. 
https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html

Reply:
Done! I've omitted Cohere models in `BedrockChatProvider` for now, as we don't have a way to set `beta_use_converse_api=true` from the Settings UI right now.